mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/util/HierarchyScanner.java

```diff
@@ -0,0 +1,191 @@
+   * @return The type hierarchy, ordered from typeElement to the highestAncestor, as dictated by the
+   *     resolved {@link HierarchyScanStrategy}
+   */
+  public static Set<TypeElement> resolveTypeHierarchy(TypeElement typeElement, Types types) {
+    HierarchyScanStrategy hierarchyScanStrategy =
+        HierarchyScanner.resolveHierarchyScanStrategy(typeElement, types);
+
+    ImmutableSet.Builder<TypeElement> hierarchy = ImmutableSet.builder();
+    traverseFullHierarchy(hierarchyScanStrategy, typeElement, types, hierarchy::add);
+    return hierarchy.build();
+  }
+
+  private static HierarchyScanStrategy resolveHierarchyScanStrategy(
+      TypeElement classElement, Types types) {
+    // Use the default HierarchyScanStrategy to find the configured HierarchyScanStrategy.
+    // This is done because the default strategy is the most permissive.
+    HierarchyScanStrategy strategy =
+        ClassForDefaultScanStrategy.class.getAnnotation(HierarchyScanStrategy.class);
+
+    // traverse hierarchy until a strategy is found.
+    final AtomicReference<HierarchyScanStrategy> ref = new AtomicReference<>(strategy);
+    traverseHierarchy(
+        strategy,
+        classElement,
+        types,
+        (TypeElement t) -> {
+          HierarchyScanStrategy discoveredStrategy = t.getAnnotation(HierarchyScanStrategy.class);
+          // if we find a strategy, set it and stop traversing.
+          if (discoveredStrategy != null) {
+            ref.compareAndSet(strategy, discoveredStrategy);
+            return false;
+          }
+          return true;
+        });
+
+    return ref.get();
+  }
+
+  private static void traverseFullHierarchy(
+      HierarchyScanStrategy hierarchyScanStrategy,
+      TypeElement classElement,
+      Types types,
+      Consumer<TypeElement> typeConsumer) {
+    traverseHierarchy(
+        hierarchyScanStrategy,
+        classElement,
+        types,
+        (TypeElement t) -> {
+          typeConsumer.accept(t);
+          return true;
+        });
+  }
+
+  private static void traverseHierarchy(
+      HierarchyScanStrategy hierarchyScanStrategy,
+      TypeElement classElement,
+      Types types,
+      Function<TypeElement, Boolean> typeConsumer) {
+
+    if (!typeConsumer.apply(classElement)) {
+      return;
+    }
+
+    Set<TypeMirror> interfacesToScan = Collections.emptySet();
+    // TODO: Probably should also consider interfaces for highest class
+    boolean atHighestClass =
+        classElement
+            .getQualifiedName()
+            .toString()
+            .equals(hierarchyScanStrategy.highestAncestor().getName());
+    while (!atHighestClass) {
+      // add super class
+      TypeMirror superClass = classElement.getSuperclass();
+      TypeElement superClassElement = null;
+      if (superClass.getKind() == TypeKind.DECLARED) {
+        superClassElement = (TypeElement) types.asElement(superClass);
+        atHighestClass =
+            superClassElement
```

Comment:
I think that you can simplify that check with the usage of `ProcessorContext.Classes.isSame()`

Reply:
nice, that's way better, and more correct. will use that.